authors/apps/authentication/email_verification.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.utils import six
+
+from django.http import HttpResponse
+from django.shortcuts import render 
+from django.contrib.sites.shortcuts import get_current_site
+from django.utils.encoding import force_bytes, force_text
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django.template.loader import render_to_string
+from django.contrib.auth.models import User
+from django.core.mail import EmailMessage
+
+class TokenGenerator(PasswordResetTokenGenerator):
+    #This funtion used the PasswordResetTokenGenerator in django. 
+    #You can find out how it works here "https://shoop.readthedocs.io/en/v4.0.0/_modules/django/contrib/auth/tokens.html"
+    def _make_hash_value(self, user, timestamp):
+        return (
+            six.text_type(user.pk) + six.text_type(timestamp) +
+            six.text_type(user.is_active)
+        )
+account_activation_token = TokenGenerator()
+
+class SendActivationEmail():
+    def send_account_activation_email(self, request, email):
+        user = User.objects.filter(email=email).first()
+        if user:
+            current_site = get_current_site(request)
+            email_subject = 'Activate your Authors Haven account.'
+            email_message = render_to_string('templates/email_verification.html', {
+                'user': user,
+                'domain': current_site.domain,
+                'uid':urlsafe_base64_encode(force_bytes(user.pk)),
+                'token':account_activation_token.make_token(user),
+            })
+            user_email = user.email
+            email = EmailMessage(
+                        email_subject, email_message, to=[user_email]
+            )
+            email.send()
+            #i'm still looking for a way to return the token (account_activation_token) here
```

Comment:
You can create an activation link this way,
`activation_link = 'http://' + current_site_domain + 'api/account/activate/' + token + '/'` and send it via email
then create `/account/activate` endpoint to verify the user when he/she click the link

Reply:
Thanks a lot @charisschomba